### ISSUE TITLE
NULLs are not supported when writing to protobuf

### ISSUE THREAD

**Issue by sandwwraith** [Kotlin/kotlinx.serialization#71]
https://github.com/Kotlin/kotlinx.serialization/issues/58#issuecomment-359719900

```kotlin
@Serializable class SampleMessage {
    @Optional @SerialId(1) var name: String? = null
}

fun main(args: Array<String>) {
    val message = SampleMessage()
    println(ProtoBuf.dumps(message))
}
```

crashes with 

```
Exception in thread "main" kotlinx.serialization.SerializationException: null is not supported
	at kotlinx.serialization.TaggedOutput.writeTaggedNull(Tagged.kt:38)
	at kotlinx.serialization.TaggedOutput.writeNullValue(Tagged.kt:80)
	at kotlinx.serialization.KOutput.writeNullableSerializableValue(Serialization.kt:151)
	at kotlinx.serialization.KOutput.writeNullableSerializableElementValue(Serialization.kt:198)
	at SampleMessage.write$Self(TestOneOf.kt)
	at SampleMessage$$serializer.save(TestOneOf.kt)
	at SampleMessage$$serializer.save(TestOneOf.kt:5)
	at kotlinx.serialization.KOutput.write(Serialization.kt:99)
	at kotlinx.serialization.protobuf.ProtoBuf.dump(ProtoBuf.kt:415)
	at TestOneOfKt.main(TestOneOf.kt:17)
```

**Comment by wem** [comment]
Any news on this issue? Unfortunately, this bug makes Protobuf serialization unusable at the moment.

**Comment by sandwwraith** [comment]
@wem It took longer than I though. I think it would be resolved alongside with #58 , but with one limitation: protobuf doesn't have a concept of `null` anyway, so the single way to record them is to just skip them in output stream. But when you have an optional value which default is not a null value, you can't really say was it missing from the stream because it had default value or because it was null, so such declaration will likely be prohibited by protobuf.

**Comment by raderio** [comment]
I think this https://github.com/Kotlin/kotlinx.serialization/issues/407 may also be a solution
